cmd/ls.go

```diff
@@ -129,3 +145,82 @@ func init() {
 
 	lsCmd.Flags().BoolP("long", "l", false, "Long listing")
 }
+
+func listOfEntryNames(entries []*files.Metadata) []string {
+	listOfEntryNames := []string{}
+
+	for _, entry := range entries {
+		switch entry.Tag {
+		case "folder":
+			listOfEntryNames = append(listOfEntryNames, entry.Folder.Name+"    ")
+		case "file":
+			listOfEntryNames = append(listOfEntryNames, entry.File.Name+"    ")
+		}
+	}
+
+	sort.Strings(listOfEntryNames)
+	return listOfEntryNames
+}
+
+func lengthOfEntryNames(listOfEntryNames []string) []int {
+	lengths := []int{}
+	for _, entry := range listOfEntryNames {
+		lengths = append(lengths, len(entry))
+	}
+	return lengths
+}
+
+func fewestColumns(lengths []int) int {
+	termWidth, _ := terminalWidth()
+
+	columns := 1
+	prevLength := 0
+	for _, length := range lengths {
+		if length+prevLength+4 < termWidth {
+			columns++
+			prevLength += length + 4
+		} else {
+			break
+		}
+	}
+	return columns
+}
+
+func columnLength(lengths []int) int {
+	sort.Ints(lengths)
+	return reverse(lengths)[0] + 4
+}
+
+func resizeEntries(fullSize int, entryNames []string) []string {
+	for i, entry := range entryNames {
+		for len(entry) != fullSize {
+			entry += " "
+		}
+		entryNames[i] = entry
+	}
+	return entryNames
+}
+
+func numberOfRows(numberOfRows, numberOfNames int) int {
+	return int(math.Ceil(float64(numberOfNames) / float64(numberOfRows)))
+}
+
+func reverse(input []int) []int {
+	if len(input) == 0 {
+		return input
+	}
+	return append(reverse(input[1:]), input[0])
+}
+
+func terminalWidth() (int, error) {
+	sttyCmd := exec.Command("stty", "size")
```

Comment:
I'd prefer not to fork out if possible. Can't we use something like https://godoc.org/golang.org/x/crypto/ssh/terminal#GetSize?